src/XRP/xrp-path-element.ts

```diff
@@ -0,0 +1,29 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrency from './xrp-currency'
+
+type PathElement = Payment.PathElement
+
+/*
+ * A path step in an XRP Ledger Path.
+ * @see: https://xrpl.org/paths.html#path-steps
+ */
+export default class XRPPathElement {
+  public static from(pathElement: PathElement): XRPPathElement {
+    const account = pathElement.getAccount()?.getAddress()
+    const currency = pathElement.getCurrency()
+    let xrpCurrency
+    if (currency) {
+      xrpCurrency = XRPCurrency.from(currency)
```

Comment:
Is it worth making the `from()` return `undefined` if given `undefined`? @keefertaylor

Reply:
I, personally, feel that makes APIs less specific. That is to say `string|undefined` (optional string) is harder to reason about than `string` (string). 

While it would provide neat syntactical sugar at this level, I'm not sure the cost outweighs the benefit. Happy to hear if you feel strongly.

Reply:
Ok, I'm going to leave this as it is for now, and if we decide to change later we can do them all at once.

Reply:
Ok, I do have one suggestion.

Instead of:
```js
let xrpCurrency
if (currency) {
  xrpCurrency = XRPCurrency.from(currency)
} else {
  xrpCurrency = undefined
}
```

I think you can write that as:
```js
const xrpCurrency = currency && XRPCurrency.from(currency)
```

This works because your `if` statement is only asserting that `currency` is truthy. I forget what exactly this usage of the `&&` operator is called, but basically, it should function the same as your `if/else` above.

Reply:
nice!! cleaner.  I think there are a few places where this type of logic appears while converting from protos to native objects - I'll take another pass on the rest of this PR chain as well.